src/ristretto/dalek_range_proof.rs

```diff
@@ -130,6 +188,63 @@ mod test {
         assert_eq!(prover.verify(&proof, &c), false);
     }
 
+    #[test]
+    fn create_and_rewind_proof() {
+        let base = PedersenCommitmentFactory::default();
+
+        let prover = DalekRangeProofService::new(1 << 5, &base).unwrap();
+        let mut rng = thread_rng();
+        let k = RistrettoSecretKey::random(&mut rng);
+        let v = RistrettoSecretKey::from(42);
+
+        let rewind_k = RistrettoSecretKey::random(&mut rng);
+        let rewind_blinding_k = RistrettoSecretKey::random(&mut rng);
+        let random_k = RistrettoSecretKey::random(&mut rng);
+
+        let public_rewind_k = RistrettoPublicKey::from_secret_key(&rewind_k);
+        let public_rewind_blinding_k = RistrettoPublicKey::from_secret_key(&rewind_blinding_k);
+        let public_random_k = RistrettoPublicKey::from_secret_key(&random_k);
+
+        let commitment_factory: PedersenCommitmentFactory = PedersenCommitmentFactory::default();
+        let c = commitment_factory.commit(&k, &v);
+        let message = b"testing123456789101";
+        let proof = prover.construct_proof_with_rewind_key(&k, 42, &rewind_k, &rewind_blinding_k, &message).unwrap();
+
+        if let Err(RangeProofError::InvalidRewind) =  prover.rewind_proof_value_only(&proof, &c, &public_random_k, &public_rewind_blinding_k) {
+            assert!(true);
+        } else {
+            assert!(false, "Should be an invalid rewind");
+        }
```

Comment:
Is `RangeProofError` impl PartialEq? If so (or add it to derive):

```
assert_eq!(prover.rewind_proof_value_only(&proof, &c, &public_random_k, &public_rewind_blinding_k), Err(RangeProofError::InvalidRewind))
```

Reply:
Will do, easy enough for this small error. The reason I don't do it out of hand these days is because in the larger error structs in the main repo you can't just derive PartialEq because the error structs are so deeply nested that almost all of them have some weird embedded struct that doesn't implement PartialEq and after all this is only needed for slightly neater tests.